### DISCUSSION TITLE
Router Level Dependencies are not injected

### DISCUSSION THREAD

**Discussion by ilterisYucel** [fastapi/fastapi#11527] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# in dependencies.py
def get_service_dependency(session: AsyncSession = Depends(get_db_session)):
    return OwnerService(session)

# in router.py
owner_router = APIRouter(prefix="/owners", tags=["owners"], dependencies=[Depends(get_service_dependency)])

@owner_router.get("/", response_model=List[GetOwnerSchema])
async def get_owners():
    try:
        service: OwnerService = owner_router.dependencies[0].dependency()
        result: List[GetOwnerSchema] = await service.get_owners()
        return result
    except Exception as e:
        print(e)
        raise HTTPException(status_code=500)
```

### Description

When i add dependency in router level and tried to use in handler function dependency is not injected and raise error
Depends object has no attribute 'xxx'.

But If i add dependecy in handler function parameter, dependency injected and i get successfull response.

Am I wrong to use dependency in router level or it is a bug? Thanks for response.

### Operating System

Linux

### FastAPI Version

0.110.2

### Pydantic Version

2.7.1

### Python Version

3.11.6

**Comment by YuriiMotov** [comment, marked as answer]
FastAPI doesn't save solved results of router-level dependencies. So, you can't access these results.
Here is a related part of FastAPI sources:
https://github.com/tiangolo/fastapi/blob/1c3e6918750ccb3f20ea260e9a4238ce2c0e5f63/fastapi/dependencies/utils.py#L524

If you need to use the result of dependency, you should include this dependency as a parameter of endpoint function
